docs/DeveloperGuide.md

```diff
@@ -1,34 +1,610 @@
+ 
+Step 5. The Zoom event is then added to the user's `UserData` for further use.
+ 
+The following sequence diagram shows how the whole add feature works: <br>
+
+![Sequence Diagram for Add Command](./diagrams/addCommand.jpg)
+
+#### List feature
+
+The list feature allows the user to print a list of events added by type.
+The list of events is print according to the order it was added in.
+
+User calls the list command by executing `list [argument]`.
+
+Executing `list` without specifying any argument prints a list of event types available.  
+Specifying an event type as the argument prints a list of events of that type.  
+Specifying `all` as the argument prints a list of all events in the order: Personal, Timetable, Zoom.
+
+Given below is an example usage scenario of the list feature.
+
+Step 1. The user executes `list zoom` command to print a list of Zoom events.
+The `list` command is passed through a parser to return the ListCommand with arguments initialised.
+
+Step 2. `ListCommand#execute()` is called, retrieving the list of Zoom events in `UserData`.
+
+Step 3. `ListCommand#execute()` calls `Ui#printList()` and passes the list of Zoom events to print.
+
+Step 4. `Ui#printList()` loops through every event in the Zoom event arraylist and prints it.
+
+The following sequence diagram shows how the `list zoom` command is parsed:
+
+![Sequence diagram for parsing list command](./diagrams/ListParseSequenceDiagram.jpg)
+
+#### Calendar feature
+(WIP)
+
+#### Repeat feature
+
+The repeat feature on the program allows for the user to be able to make certain events repeat several times over a defined time period.
+For instance, the user can request for a personal event to be repeated monthly for four months. To run the program, the user will need to key in the command `repeat [event type] [index] [timeUnit] [count]`
+
+|Argument| Description |
+|--------|----------|
+|event type|What type of event is to be repeated? Accepted arguments are `personal`, `timetable` or `zoom`|
+|index|Index number of the event to be repeated that is stored on the Event List|
+|timeUnit|For what unit of time each event is to be repeated. Accepted arguments are `daily`, `weekly` and `monthly`|
+|count|Integer indicating how many times the event is to be repeated|
+
+For example, to repeat a personal event located at index 2 for 3 days, the user will key in this: `repeat personal 2 daily 3`.
+
+The following sequence diagram shows the overall process of repeating an event:
+
+![Sequence Diagram for Repeat Command](./diagrams/RepeatScenario.jpg)
+
+The steps to repeating the command will be described in the following sections. 
+
+##### Step 1: Obtaining the event information
+
+The repeat command first locates the event that is to be repeated along with its starting date.
+Next, it creates an empty ArrayList called `repeatEventList` that will contain all repeated events.
+
+![Sequence Diagram for Repeat Command step 1](./diagrams/repeatstep1.jpg)
```

Comment:
Are the arrow heads and tails supposed to connect here?

![image](https://user-images.githubusercontent.com/1352844/97534151-77597c80-19f4-11eb-8e82-99ccf7c050e3.png)